File: app/src/main/java/com/thalesgroup/gemalto/fido2/sample/domain/logger/LoggerImpl.java
package com.thalesgroup.gemalto.fido2.sample.domain.logger;

import java.util.ArrayList;
import java.util.List;

public class LoggerImpl implements Logger {

    private List<String> items;

    public LoggerImpl() {
        items = new ArrayList<>();
    }

    @Override
    public void log(String text) {
        items.add(text);
    }

    @Override
    public List<String> getLogs() {
        return items;
    }

    @Override
    public void clean() {
        items.clear();
    }


}

File: app/fido2sample/src/main/kotlin/com/thalesgroup/gemalto/fido2/sample/domain/logger/LoggerImpl.kt
/*
 * Copyright © 2021 THALES. All rights reserved.
 */
package com.thalesgroup.gemalto.fido2.sample.domain.logger

open class LoggerImpl : Logger {
    override val logs: MutableList<String?>

    init {
        this.logs = ArrayList<String?>()
    }

    override fun log(text: String?) {
        logs.add(text)
    }

    override fun clean() {
        logs.clear()
    }
}
